ClosedXML/Excel/NamedRanges/XLNamedRange.cs

```diff
@@ -9,24 +10,40 @@ internal class XLNamedRange : IXLNamedRange
         private String _name;
         private readonly XLNamedRanges _namedRanges;
 
+        internal XLWorkbook Workbook => _namedRanges.Workbook;
+
         public XLNamedRange(XLNamedRanges namedRanges, String rangeName, String range, String comment = null)
+            : this(namedRanges, rangeName, comment)
         {
-            _namedRanges = namedRanges;
-            Visible = true;
-            Name = rangeName;
             RangeList.Add(range);
-            Comment = comment;
         }
 
         public XLNamedRange(XLNamedRanges namedRanges, String rangeName, IXLRanges ranges, String comment = null)
+            : this(namedRanges, rangeName, comment)
         {
-            _namedRanges = namedRanges;
+            ranges.ForEach(r => RangeList.Add(r.RangeAddress.ToStringFixed(XLReferenceStyle.A1, true)));
+        }
+
+        private XLNamedRange(XLNamedRanges namedRanges, String rangeName, String comment)
+        {
+            _namedRanges = namedRanges ?? throw new ArgumentNullException(nameof(namedRanges));
             Visible = true;
             Name = rangeName;
-            ranges.ForEach(r => RangeList.Add(r.RangeAddress.ToStringFixed(XLReferenceStyle.A1, true)));
             Comment = comment;
         }
 
+        /// <summary>
+        /// Checks if the named range contains invalid references (#REF!).
+        /// </summary>
+        public bool IsValid
+        {
+            get
+            {
+                return RangeList.SelectMany(c => c.Split(',')).All(r =>
```

Comment:
Side note, I think we should start thinking of storing the ranges in something else than a `List<String>` for better performance. To do in future issue, not this one.